src/influx_line.erl

```diff
@@ -0,0 +1,626 @@
+    encode(
+      [
+        [
+          {measurement, <<"test">>},
+          {fields, #{"val" => 10}},
+          {time, 1}
+        ],
+        [
+          {measurement, <<"test2">>},
+          {fields, #{"val" => 20}},
+          {time, 2}
+        ]
+      ]
+    )
+  ).
+
+encode_map_without_measurement_test() ->
+  ?assertEqual(
+    {error, invalid_data},
+    encode(
+      #{
+        fields => #{ val => 10}
+      }
+    ) 
+  ).
+
+encode_map_without_fields_test() ->
+  ?assertEqual(
+    {error, invalid_data},
+    encode(
+      #{
+        measurement => test,
+        tags => #{ val => 10}
+      }
+    ) 
+  ).
+
+encode_proplist_without_measurement_test() ->
+  ?assertEqual(
+    {error, invalid_data},
+    encode(
+      [
+        {fields,  [{val, 10}] }
+      ]
+    ) 
+  ).
+
+encode_proplist_without_fields_test() ->
+  ?assertEqual(
+    {error, invalid_data},
+    encode(
+      [
+        {measurement, test},
+        {tags, #{ val => 10}}
+      ]
+    ) 
+  ).
+
+encode_at_least_one_invalid_test() ->
+  ?assertEqual(
+    {error, invalid_data},
+    encode(
+      [
+        [
+          {measurement, test},
+          {fields, #{ val => 10}}
+        ],
+        [
+          {tags, #{ val => 10}}
+        ]
+      ]
+    ) 
+  ).
+
+encode_3_with_proplist_tags_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west,ip=1.1.1.1 1=1">>,
+    encode(<<"test">>, #{1 => 1}, [{host, "eu-west"}, {ip, '1.1.1.1'}])
+  ).
+
+encode_3_with_map_tags_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west,ip=1.1.1.1 1=1">>,
+    encode(<<"test">>, #{1 => 1}, #{host => 'eu-west', ip => <<"1.1.1.1">>})
+  ).
+
+encode_3_with_binary_tags_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west,ip=1.1.1.1 1=1">>,
+    encode(<<"test">>, #{1 => 1}, <<",host=eu-west,ip=1.1.1.1">>)
+  ).
+
+encode_3_many_points_with_tags_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west,num=111 memory=\"high\",cpu=20 100\ntest,host=eu-west,num=111 memory=\"low\",cpu=30 101\n">>,
+    encode(
+      <<"test">>,
+      [
+        [{memory, "high"}, {"cpu", 20}],
+        [{<<"memory">>, "low"}, {cpu, 30}]
+      ], 
+      #{host => 'eu-west', num => 111},
+      100
+    )
+  ).
+
+encode_4_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west,ip=1.1.1.1 1=1 123123">>,
+    encode(<<"test">>, #{1 => 1}, [{host, "eu-west"}, {ip, '1.1.1.1'}], 123123)
+  ).
+
+encode_4_many_points_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west,ip=1.1.1.1 1=1,text=\"hello\" 123123">>,
+    encode(<<"test">>,
+      [[{"1", 1}, {text, <<"hello">>}]],
+      [{host, "eu-west"}, {ip, '1.1.1.1'}],
+      123123
+    )
+  ).
+
+encode_escape_tags_test() ->
+  ?assertEqual(
+    <<"test,host=eu\\ we\\=st,ip=1\\,1\\,1\\,1 1=1">>,
+    encode(<<"test">>,
+      [[{"1", 1}]],
+      [{host, "eu we=st"}, {ip, '1,1,1,1'}]
+    )
+  ).
+
+encode_escape_fields_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west 1=1,text=\"hello\\ \\\"man\\=\\=human\\,\\ yo!\\\"\"">>,
```

Comment:
Line 619 is too long:     <<"test,host=eu-west 1=1,text=\"hello\\ \\\"man\\=\\=human\\,\\ yo!\\\"\"">>,.